src/provider/element-templates/util/templateUtil.js

```diff
@@ -14,18 +15,24 @@ export function unlinkTemplate(element, injector) {
 
 export function removeTemplate(element, injector) {
   const replace = injector.get('replace'),
-        selection = injector.get('selection');
+        selection = injector.get('selection'),
+        modeling = injector.get('modeling');
 
   const businessObject = getBusinessObject(element);
 
   const type = businessObject.$type,
-        eventDefinitionType = getEventDefinitionType(businessObject);
+        eventDefinitionType = getEventDefinitionType(businessObject),
+        label = getLabel(element);
 
   const newElement = replace.replaceElement(element, {
     type: type,
     eventDefinitionType: eventDefinitionType
   });
 
+  if (label) {
+    modeling.updateLabel(newElement, label);
```

Comment:
As we are firing a command here, the next undo after removing the template will remove the label

![Kapture 2022-04-11 at 11 40 51](https://user-images.githubusercontent.com/9433996/162711798-d795a3c2-03b8-4b59-b6fc-a35b9116ca11.gif)

Only after the second undo the remove is undone. I'd expect that `undo` after the removal brings me back the template immediately.

Reply:
See also https://github.com/bpmn-io/bpmn-js-properties-panel/pull/662#pullrequestreview-937755048.